packages/jest.config.client.js

```diff
@@ -0,0 +1,12 @@
+module.exports = {
+  displayName: 'package-client',
+  rootDir: '../',
+  setupTestFrameworkScriptFile: '<rootDir>/packages/jest-setup.client.js',
+  testMatch: ['<rootDir>/packages/**/client/**/*.test.js'],
+  transformIgnorePatterns: ['/node_modules/(?!@?pubsweet|xpub)'],
+  moduleNameMapper: {
+    '\\.s?css$': 'identity-obj-proxy',
+    '^ui(.*)$': '<rootDir>/app/components/ui$1',
+    '^global(.*)$': '<rootDir>/app/components/global$1',
+  },
+}
```

Comment:
Not sure I'm a fan of config files like this living outside of components. If the component itself has tests, I feel that the component should have all the files and modules necessary for it to run as it's own package.

I'd rather have the jest files duplicated across the different packages for the above reason.

Reply:
A fair point, we put these here to make it as easy as possible to add new components in without the need to create new config and set-up files for both the server and client for every component. This creates a problem though as context specific test set up and config requirements will need to be placed in these top level files, so we'll no longer have our contexts completely stand alone. 

Just a note though, `ui` and `global` shouldn't exist anymore after the refactor as they'll sit in a `@elifesciences/component-elife-ui` package of their own.

Reply:
Does this mean we should move these first then?

Reply:
@cooryd is this what you mean: 

if we have jest setup and jest config files inside each component's client and server folder, then the issue is that the rootDir is then at that package's level.

So for instance, if we have in package.json:
```
"jest": {
//...
   projects: [
      //...,
      "<rootDir>/packages/component-dashboard/client/jest.config.js",
      //...
   ]
}
```

then the rootDir becomes elife-xpub/packages/component-dashboard

which then breaks the moduleNameMapper setting as that uses rootDir, so we would need to add some '../' but thats doable until ui and global go to their own component?

Reply:
I've done a quick test with the following config in packages/component-dashboard/client/jest.config.js:

```
module.exports = {
    displayName: 'package-client',
    rootDir: '../',
    setupTestFrameworkScriptFile: '<rootDir>/client/jest-setup.js',
    testMatch: ['<rootDir>/client/**/*.test.js'],
    transformIgnorePatterns: ['/node_modules/(?!@?pubsweet|xpub)'],
    moduleNameMapper: {
      '\\.s?css$': 'identity-obj-proxy',
      '^ui(.*)$': '<rootDir>/../../app/components/ui$1',
      '^global(.*)$': '<rootDir>/../../app/components/global$1'
    }
  }
```

seems to work

Reply:
Yes, I was trying it out locally and it seems to work. Happy for this to go into a separate issue/PR.

Reply:
I've created a `/component-dashboard/config` directory that holds the 2 config and 2 setup files for the component. Tests all pass for me and keeps the config in a central place within the component